test/api_test.js

```diff
@@ -28,31 +30,36 @@ describe("Api", () => {
 
   describe("#constructor", () => {
     it("should check that `remote` is a string", () => {
-      expect(() => new Api(42))
+      expect(() => new Api(42, events))
         .to.Throw(Error, /Invalid remote URL/);
     });
 
     it("should validate `remote` arg value", () => {
-      expect(() => new Api("http://nope"))
+      expect(() => new Api("http://nope", events))
         .to.Throw(Error, /The remote URL must contain the version/);
     });
 
     it("should strip any trailing slash", () => {
-      expect(new Api(`http://test/${SPV}/`).remote).eql(`http://test/${SPV}`);
+      expect(new Api(`http://test/${SPV}/`, events).remote).eql(`http://test/${SPV}`);
+    });
+
+    it("should expose an events property", () => {
+      expect(new Api(`http://test/${SPV}`, events).events)
+        .to.be.an.instanceOf(EventEmitter);
     });
 
     it("should assign version value", () => {
-      expect(new Api(`http://test/${SPV}`).version).eql(SPV);
-      expect(new Api(`http://test/${SPV}/`).version).eql(SPV);
+      expect(new Api(`http://test/${SPV}`, events).version).eql(SPV);
+      expect(new Api(`http://test/${SPV}/`, events).version).eql(SPV);
     });
 
     it("should accept a headers option", () => {
-      expect(new Api(`http://test/${SPV}`, {headers: {Foo: "Bar"}}).optionHeaders)
+      expect(new Api(`http://test/${SPV}`, events, {headers: {Foo: "Bar"}}).optionHeaders)
         .eql({Foo: "Bar"});
     });
 
     it("should validate protocol version", () => {
-      expect(() =>new Api(`http://test/v999`))
+      expect(() =>new Api(`http://test/v999`, events))
```

Comment:
Yeah, I'll update to handle passed the `events` instance as an option for all classes.